src/foam/java/Enum.js

```diff
@@ -1,106 +1,166 @@
-// Test with: foam.u2.ControllerMode.model_.javaSource();
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 foam.CLASS({
-  refines: 'foam.core.EnumModel',
+  package: 'foam.java',
+  name: 'EnumModel',
+  extends: 'foam.java.Class',
 
-  methods: [
+  properties: [
+    ['javaType', 'java.lang.Enum'],
+    ['javaInfoType', 'foam.core.AbstractFObjectPropertyInfo'],
+    ['javaJSONParser', 'foam.lib.json.FObjectParser'],
   ],
+  methods: [
+    function outputJava(o) {
```

Comment:
This is a good start.

Next I would look at if you can refactor this to share some code with outputJava in foam.java.Class.  There is a fair amount of duplication here that can probably be cleaned up.